CaliptraChecklistAndEvaluationMethodology.md

```diff
@@ -0,0 +1,303 @@
+
+* **Checklist Item:**  
+  * **Requirement:** The pathways through which external entities interact with Caliptra MUST be securely implemented, ensuring that only authorized interactions are permitted, and that these interactions do not compromise Caliptra's security.  
+  * **Evaluation Methodology:** Manufacturers MUST outline the architecture of interactions between external entities and Caliptra, emphasizing access controls, authentication mechanisms, and any intermediaries like security processors. Details on how untrusted entities are prevented from accessing Caliptra internals must be provided.
+
+## Specific Functionality
+
+### *Caliptra Mode Selection and State Handling*
+
+* **Checklist Item:**  
+  * **Requirement:** Where SoC behavior necessary to meet other requirements in this document is conditioned on SoC lifecycle state, the SoC MUST ensure that its own lifecycle is in a state that meets the requirements here prior to transitioning Caliptra's operational mode from unprovisioned to manufacturing or production.  
+  * **Evaluation Methodology:** Manufacturers MUST provide a state machine diagram or equivalent documentation showing how Caliptra's mode transitions are managed and synchronized with the SoC's states.
+
+### *Error Handling*
+
+* **Checklist Item:**  
+  * **Requirement:** The SoC MUST correctly implement error handling and reporting mechanisms for interactions with Caliptra, especially for security-relevant events, without leaking sensitive information.  
+  * **Evaluation Methodology:** Manufacturers MUST document the error handling procedures, including what information is reported, how errors are logged, and measures taken to prevent information leakage. Specific error codes and messages should be reviewed for appropriateness.
+
+### *Log Functionality in SoC*
+
+* **Checklist Item:**  
+  * **Requirement:** The SoC SHOULD implement logging functionality that records relevant events related to Caliptra operations, ensuring logs are tamper-evident and provide necessary information for attestation. The log does not need to be kept secure but should be tamper-evident to detect manipulation.  
+  * **Evaluation Methodology:** Manufacturers SHOULD describe the logging mechanism, including how logs are structured, how they are protected against tampering (e.g., using cryptographic hash extensions), and how they integrate with Caliptra's measurement processes.
+
+## Secure Processes
+
+### Development Process
+
+#### *Integrity of Firmware and Hardware*
+
+* **Checklist Item:**  
+  * **Requirement:** The Caliptra RTL MUST match an official release. Netlists, layout, and other development collateral derived from the RTL MUST maintain the same behavior as the released RTL.  
+  * **Evaluation Methodology:** Manufacturers MUST provide evidence of hash verification against official releases. Manufacturers MUST provide evidence that collateral derived from the released RTL has not had its behavior modified.
```

Comment:
As discussed, is a hash verification really possible? If not, is a failure in hash verification allowed? If yes, what alternative requirement has a developer to comply with?